motor_controller/motorbench-test-project/motorbench-test-project/rb_library/rb_pwm.c

```diff
@@ -1,9 +1,43 @@
 #include "rb_pwm.h"
-#include "pwm_hs/pwm.h"
+
 #include "pwm_hs/pwm_hs_types.h"
 #include "parameters/hal_params.h"
 #include "hal/hardware_access_functions.h"
 
+/**
+ * Frixed Frequency Sine PWM Variables
+ */
+uint16_t SineDutyCycle[297] = {2500, 2548, 2596, 2644, 2692, 2740, 2788, 2836, 2883, 2931, 2978, 3025, 3071, 3118, 3164, 3210, 3255,
+3301, 3346, 3390, 3434, 3478, 3521, 3564, 3606, 3648, 3689, 3730, 3770, 3810, 3849, 3887, 3925,
+3962, 3999, 4035, 4070, 4104, 4138, 4171, 4203, 4235, 4266, 4296, 4325, 4353, 4381, 4407, 4433,
+4458, 4482, 4505, 4528, 4549, 4569, 4589, 4608, 4625, 4642, 4658, 4672, 4686, 4699, 4711, 4722,
+4732, 4740, 4748, 4755, 4761, 4766, 4770, 4772, 4774, 4775, 4775, 4773, 4771, 4768, 4764, 4758,
+4752, 4744, 4736, 4727, 4716, 4705, 4693, 4679, 4665, 4650, 4634, 4616, 4598, 4579, 4559, 4538,
+4517, 4494, 4470, 4446, 4420, 4394, 4367, 4339, 4310, 4281, 4250, 4219, 4187, 4155, 4121, 4087,
+4052, 4017, 3981, 3944, 3906, 3868, 3829, 3790, 3750, 3710, 3669, 3627, 3585, 3542, 3499, 3456,
+3412, 3368, 3323, 3278, 3233, 3187, 3141, 3095, 3048, 3001, 2954, 2907, 2859, 2812, 2764, 2716,
+2668, 2620, 2572, 2524, 2476, 2428, 2380, 2332, 2284, 2236, 2188, 2141, 2093, 2046, 1999, 1952,
+1905, 1859, 1813, 1767, 1722, 1677, 1632, 1588, 1544, 1501, 1458, 1415, 1373, 1331, 1290, 1250,
+1210, 1171, 1132, 1094, 1056, 1019, 983, 948, 913, 879, 845, 813, 781, 750, 719, 690,
+661, 633, 606, 580, 554, 530, 506, 483, 462, 441, 421, 402, 384, 366, 350, 335,
+321, 307, 295, 284, 273, 264, 256, 248, 242, 236, 232, 229, 227, 225, 225, 226,
+228, 230, 234, 239, 245, 252, 260, 268, 278, 289, 301, 314, 328, 342, 358, 375,
+392, 411, 431, 451, 472, 495, 518, 542, 567, 593, 619, 647, 675, 704, 734, 765,
+797, 829, 862, 896, 930, 965, 1001, 1038, 1075, 1113, 1151, 1190, 1230, 1270, 1311, 1352,
+1394, 1436, 1479, 1522, 1566, 1610, 1654, 1699, 1745, 1790, 1836, 1882, 1929, 1975, 2022, 2069,
+2117, 2164, 2212, 2260, 2308, 2356, 2404, 2452};
+
+volatile uint16_t RunningStateCounter = 0;
```

Comment:
Would like to see these packaged into a struct and initialized using a function

Reply:
did it